src/binary_search_tree.js

```diff
@@ -0,0 +1,192 @@
+class Node {
+  constructor() {
+    this.parent = null
+    this.rightChild = null
+    this.leftChild = null
+    this.data
+  }
+}
+//////HELPER FUNCTIONS
+function hasChildren(node) {
+  return node.rightChild || node.leftChild
+}
+
+function recursiveInsert(startNode,newNode) {
+  if(newNode.data<=startNode.data) {
+    if(!startNode.leftChild) {
```

Comment:
This code would be more readable if you could extract these operations into their own functions.
Example, you are inserting into the `left subtree` and `right subtree`. 
So you can have a function like `addToRightTree` or `addToLeftTree`